common/models/access-token.js

```diff
@@ -143,8 +143,12 @@ module.exports = function(AccessToken) {
       assert(this.ttl >= -1, 'token.ttl must be >= -1');
 
       var AccessToken = this.constructor;
-      var userRelation = AccessToken.relations.user; // may not be set up
-      var User = userRelation && userRelation.modelTo;
+      var User = AccessToken.registry.getModelByType('User');
```

Comment:
unless i get the getModelByType behavior wrong (cf. [doc](https://apidocs.strongloop.com/loopback/v/3.2.1/#loopback-getmodelbytype-1)), this method 
> can be used by LoopBack to find configured models in models.json **over the base model**

in other words in the case you describe, the custom User model should be fetched correctly by `AccessToken.registry.getModelByType('User');`

however as it is equivalent, i switched back to the original implementation you are more confortable with